libs/service/api_service.cpp

```diff
@@ -432,6 +442,206 @@ ebpf_verify_and_load_program(
+        return std::string();
+    }
+    return std::string(issuer.data());
+}
+
+_Must_inspect_result_ ebpf_result_t
+ebpf_verify_sys_file_signature(
+    _In_z_ const wchar_t* file_name,
+    _In_z_ const char* issuer_name,
+    _In_ size_t eku_count,
+    _In_reads_(eku_count) const char** eku_list)
+{
+    ebpf_result_t result = EBPF_OBJECT_NOT_FOUND;
+    EBPF_LOG_ENTRY();
+    std::string required_issuer(issuer_name);
+    std::set<std::string> required_eku_set;
+
+    if (_ebpf_service_test_signing_enabled) {
+        // Test signing is enabled, so we don't verify the signature.
+        EBPF_RETURN_RESULT(EBPF_SUCCESS);
+    }
+
+    for (size_t i = 0; i < eku_count; i++) {
+        required_eku_set.insert(eku_list[i]);
+    }
+
+    try {
+        WinVerifyTrustHelper wrapper(file_name);
+
+        for (DWORD i = 0; i < wrapper.cert_count(); i++) {
+
+            std::set<std::string> eku_set = _ebpf_extract_eku(wrapper.get_cert(i));
+            std::string issuer = _ebpf_extract_issuer(wrapper.get_cert(i));
+
+            if (issuer != required_issuer) {
+                EBPF_LOG_MESSAGE_STRING(
+                    EBPF_TRACELOG_LEVEL_ERROR,
+                    EBPF_TRACELOG_KEYWORD_API,
+                    "Certificate issuer mismatch",
+                    issuer.c_str());
+                continue;
+            }
+
+            std::set<std::string> eku_intersection;
+            std::set_intersection(
+                eku_set.begin(),
+                eku_set.end(),
+                required_eku_set.begin(),
+                required_eku_set.end(),
+                std::inserter(eku_intersection, eku_intersection.begin()));
+
+            if (eku_intersection.size() != required_eku_set.size()) {
+                EBPF_LOG_MESSAGE_STRING(
+                    EBPF_TRACELOG_LEVEL_ERROR,
+                    EBPF_TRACELOG_KEYWORD_API,
+                    "Certificate EKU mismatch",
+                    "Required EKUs not found in certificate");
+                continue;
+            }
+
+            // The certificate is valid and has the required EKUs.
+            result = EBPF_SUCCESS;
+            break;
+        }
+    } catch (const std::bad_alloc&) {
+        result = EBPF_NO_MEMORY;
+        EBPF_LOG_MESSAGE_ERROR(
+            EBPF_TRACELOG_LEVEL_ERROR,
+            EBPF_TRACELOG_KEYWORD_API,
+            "Memory allocation failed during signature verification",
+            result);
+    } catch (const std::runtime_error&) {
+        result = EBPF_INVALID_ARGUMENT;
+        EBPF_LOG_WIN32_API_FAILURE(EBPF_TRACELOG_KEYWORD_API, WinVerifyTrust);
+    }
+
+    EBPF_RETURN_RESULT(result);
+}
+
+static const char* _ebpf_required_issuer =
+    "US, Washington, Redmond, Microsoft Corporation, Microsoft Corporation eBPF Verification";
+static const char* _ebpf_required_eku_list[] = {
+    "1.3.6.1.5.5.7.3.3",     // Code Signing
+    "1.3.6.1.4.1.311.133.1", // eBPF Verification
```

Comment:
This string appears multiple times in this PR hard-coded.  Can it be in a define in a header file?